src/group_theory/schreier.lean

```diff
@@ -73,4 +73,16 @@ begin
     exact (H.mul_mem_cancel_left (hU hg)).mp hh },
 end
 
+/-- **Schreier's Lemma**: If `R` is a right_transversal of `H : subgroup G`
+  with `1 ∈ R`, and if `G` is generated by `S : set G`, then `H` is generated by the set
+  `(R * S).image (λ g, g * (mem_right_transversals.to_fun hR g)⁻¹)`. -/
+lemma closure_mul_image_eq_top
+  (hR : R ∈ right_transversals (H : set G)) (hR1 : (1 : G) ∈ R) (hS : closure S = ⊤) :
+  closure ((R * S).image (λ g, ⟨g * (mem_right_transversals.to_fun hR g)⁻¹,
+    mem_right_transversals.mul_inv_to_fun_mem hR g⟩) : set H) = ⊤ :=
```

Comment:
This statement is somewhat hard to read.
Is `g * (mem_right_transversals.to_fun hR g)⁻¹,  mem_right_transversals.mul_inv_to_fun_mem hR g` something that deserves a wrapper definition? Is there another way to express the result?

Reply:
In effect, `mem_right_transversals.mul_inv_to_fun_mem hR g : ` is the wrapper, since it says `g * (mem_right_transversals.to_fun hR g)⁻¹ ∈ H`.

One option would be to write: `closure ((R * S).image (λ g, ⟨_, mem_right_transversals.mul_inv_to_fun_mem hR g⟩) : set H) = ⊤ :=`
Another option would be to `open mem_right_transversals` at the start of the file.

Reply:
I've pushed a commit with `open mem_right_transversals`. Does it look fine now?